### PR TITLE
Game windows, changes to title, yellow color and colon

### PR DESCRIPTION
**Pull request by shprotru**
Fix for #1919

### CODE REVIEW

File: src/fheroes2/battle/battle_dialogs.cpp [hunk 1]
```diff
@@ -322,26 +321,26 @@ void Battle::Arena::DialogBattleSummary( const Result & res ) const
 
     cursor.SetThemes( Cursor::POINTER );
 
-    std::string msg;
+    std::string title, msg;
```
Comment (ihhub) [inline src/fheroes2/battle/battle_dialogs.cpp:324]: Please do not declare multiple variables at one use. Always declare one variable per line.

File: src/fheroes2/battle/battle_dialogs.cpp [hunk 2]
```diff
@@ -389,8 +388,16 @@ void Battle::Arena::DialogBattleSummary( const Result & res ) const
 
     fheroes2::Button btn_ok( pos_rt.x + 121, pos_rt.y + 410, ( conf.ExtGameEvilInterface() ? ICN::WINCMBBE : ICN::WINCMBTB ), 0, 1 );
 
-    TextBox box( msg, Font::BIG, 270 );
-    box.Blit( pos_rt.x + 25, pos_rt.y + 175 );
+    const bool isTitlePresent = title.length() > 0;
```
Comment (ihhub) [inline src/fheroes2/battle/battle_dialogs.cpp:391]: Please use `!title.empty()` instead.

File: src/fheroes2/battle/battle_dialogs.cpp [hunk 3]
```diff
@@ -389,8 +388,16 @@ void Battle::Arena::DialogBattleSummary( const Result & res ) const
...
+    if ( isTitlePresent ) {
+        TextBox box( title, Font::YELLOW_BIG, 270 );
```
Comment (ihhub) [inline src/fheroes2/battle/battle_dialogs.cpp:393]: Please make this variable `const`.

File: src/fheroes2/battle/battle_dialogs.cpp [hunk 4]
```diff
@@ -389,8 +388,16 @@ void Battle::Arena::DialogBattleSummary( const Result & res ) const
...
+        box.Blit( pos_rt.x + 25, pos_rt.y + 175 );
+    }
+
+    if ( msg.length() > 0 ) {
+        TextBox box( msg, Font::BIG, 270 );
```
Comment (ihhub) [inline src/fheroes2/battle/battle_dialogs.cpp:398]: Please make this variable `const`.

File: src/fheroes2/battle/battle_dialogs.cpp (hunk 3 above)
Comment (ihhub) [inline src/fheroes2/battle/battle_dialogs.cpp:393]: What the value 270 stand for? Is it a width of the window?

File: src/fheroes2/dialog/dialog_recrut.cpp [hunk 5]
```diff
@@ -87,10 +87,10 @@ void RedrawMonsterInfo( const fheroes2::Rect & pos, const Monster & monster, u32
     // text recruit monster
     std::string str = _( "Recruit %{name}" );
     StringReplace( str, "%{name}", monster.GetMultiName() );
-    text.Set( str, Font::BIG );
-    dst_pt.x = pos.x + ( pos.width - text.w() ) / 2;
+    TextBox recruitMonsterText( str, Font::YELLOW_BIG, BOXAREA_WIDTH );
```
Comment (ihhub) [inline src/fheroes2/dialog/dialog_recrut.cpp:90]: Let's please put all changes which are not present in the original game into separate pull request as we did not discussed them properly.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @shprotru could you please take a look into comments?

File: src/fheroes2/battle/battle_dialogs.cpp (hunk 3 above)
Comment (shprotru) [reply src/fheroes2/battle/battle_dialogs.cpp:393]: can't .Blit then, reason:
```
no instance of overloaded function "TextBox::Blit" matches the argument list and object (the object has type qualifiers that prevent a match) -- argument types are: (int, int) -- object type is: const TextBox
```

Comment (shprotru) [reply src/fheroes2/battle/battle_dialogs.cpp:393]: This is a text width, was taken here: https://github.com/ihhub/fheroes2/blob/master/src/fheroes2/battle/battle_dialogs.cpp#L392

File: src/fheroes2/battle/battle_dialogs.cpp (hunk 1 above)
Comment (shprotru) [reply src/fheroes2/battle/battle_dialogs.cpp:324]: done

File: src/fheroes2/battle/battle_dialogs.cpp (hunk 2 above)
Comment (shprotru) [reply src/fheroes2/battle/battle_dialogs.cpp:391]: done

File: src/fheroes2/battle/battle_dialogs.cpp (hunk 4 above)
Comment (shprotru) [reply src/fheroes2/battle/battle_dialogs.cpp:398]: can't call `.Blit` then, reason:
```
no instance of overloaded function "TextBox::Blit" matches the argument list and object (the object has type qualifiers that prevent a match) -- argument types are: (int, int) -- object type is: const TextBox
```

File: src/fheroes2/dialog/dialog_recrut.cpp (hunk 5 above)
Comment (shprotru) [reply src/fheroes2/dialog/dialog_recrut.cpp]: reverted back

File: src/fheroes2/battle/battle_dialogs.cpp (hunk 3 above)
Comment (ihhub) [reply src/fheroes2/battle/battle_dialogs.cpp:393]: Please declare a separate variable for this. Old code contains a lot of hardcoded values and we should avoid them as much as possible.

Comment (shprotru) [reply src/fheroes2/battle/battle_dialogs.cpp]: check, please

File: src/fheroes2/battle/battle_dialogs.cpp [hunk 6]
```diff
@@ -389,8 +396,15 @@ void Battle::Arena::DialogBattleSummary( const Result & res ) const
...
-    box.Blit( pos_rt.x + 25, pos_rt.y + 175 );
+    if ( !title.empty() ) {
+        TextBox box( title, Font::YELLOW_BIG, bsTextWidth );
+        box.Blit( pos_rt.x + bsTextXOffset, pos_rt.y + bsTextYOffset );
+    }
+
+    if ( msg.length() > 0 ) {
+        TextBox box( msg, Font::BIG, bsTextWidth );
+        box.Blit( pos_rt.x + bsTextXOffset, pos_rt.y + bsTextYOffset + ( !title.empty() ? bsTextIndent : 0 ) );
+    }
```
Comment (ihhub) [inline src/fheroes2/battle/battle_dialogs.cpp:407]: Let's do like this:
```
    int32_t messageYOffset = 0;
    if ( !title.empty() ) {
        TextBox box( title, Font::YELLOW_BIG, bsTextWidth );
        box.Blit( pos_rt.x + bsTextXOffset, pos_rt.y + bsTextYOffset );
        messageYOffset = bsTextIndent;
    }

    if ( !msg.empty() ) {
        TextBox box( msg, Font::BIG, bsTextWidth );
        box.Blit( pos_rt.x + bsTextXOffset, pos_rt.y + bsTextYOffset + messageYOffset );
    }
```
Is it easier to read?

Comment (shprotru) [reply src/fheroes2/battle/battle_dialogs.cpp]: done

**Comment by ihhub** [comment]
Hi @shprotru , I'm still keeping looking at this PR so don't think that we forget about this :) I just don't want to make a radical changes from the original (for now).

@Branikolog could you please take a look into the original issue and give your opinion?

**Comment by ihhub** [comment]
Hi @shprotru could you please do this:
- leave only the changes related to `Income` and `Buy Monsters` dialogs
- remove the rest of the code.

We need this as the original issue combines changes to new windows which are not present in the original game and changes in old dialogs which have the same style as in the original game.

We're okay to modify new windows as by styling is better but for old windows is debatable. After merging the changes only for old windows we can update the issue and you can create a new pull request once we agree on changes.

**Comment by shprotru** [comment]
@Branikolog this was as a suggestion(currently declined changes), need your approvement or decline.
Снимок экрана от 2020-11-24 22-13-28
Снимок экрана от 2020-11-24 22-19-19
Снимок экрана от 2020-11-24 22-15-03

**Comment by Branikolog** [comment]
Hi, @shprotru !
I agree with @ihhub, these improvements are debatable. 
But you know, actually, I like them. Looking at your screenshots, I can tell they're looking not bad! We were going to leave improvements the original UI to the latest release... And implementing them right now is probably a bit too early... 
Anyway. My +1 for this changes.

**Comment by LeHerosInconnu** [comment]
@Branikolog and @shprotru.

> Hi, @shprotru !
> But you know, actually, I like them. Looking at your screenshots, I can tell they're looking not bad!
> Anyway. My +1 for this changes.

I like this positivism. :)

Thank you @shprotru, for showing him the way. :D

**Comment by shprotru** [comment]
Hi, @LeHerosInconnu! I am done with `A glorious victory!`, but how to deal with `The enemy has surrendered!` and `The enemy has fled!` titles?

**Comment by LeHerosInconnu** [comment]
Hello @shprotru,

> Hi, @LeHerosInconnu! I am done with `A glorious victory!`, but how to deal with `The enemy has surrendered!` and `The enemy has fled!` titles?

Good question. :)

On the one hand, the title is a sentence with a verb and it can be a bit weird to capitalize every word in the sentence.

`The Enemy Has Surrendered!` and `The Enemy Has Fled!`

On the other hand, if all the words are not capitalized the rule is not respected and that is just as weird.

Maybe we should change the text to make it less weird:

"The Enemy Surrendered!"

and:

"The Enemy Fled!"

Instead of:

"The enemy has surrendered!"

and:

"The enemy has fled!"

What do you folks think?

**Comment by Branikolog** [comment]
Actually, I like all this sentences look in original game without "capitalization".

**Comment by shprotru** [comment]
Declined capitalization of `A glorious victory!` message

**Comment by ihhub** [comment]
Let's move all changes from the original game into separate pull request. We're mixing right now the original design and our new windows. We should separate it in order to track changes and implement them once everyone (majority) accepts them.

I think this is a proper way to move with this issue after all.

File: src/fheroes2/battle/battle_dialogs.cpp [hunk 7]
```diff
@@ -42,6 +42,14 @@
 
 namespace
 {
+    // DialogBattleSummary text related values
+    const int bsTextWidth = 270;
+    const int bsTextXOffset = 25;
+    const int bsTextYOffset = 175;
+    const int bsTextIndent = 30;
+
+    const std::string msgAGloriousVictory = "A glorious victory!";
```
Comment (ihhub) [inline src/fheroes2/battle/battle_dialogs.cpp:51]: Please use `const char *` instead.

Comment (ihhub) [reply src/fheroes2/battle/battle_dialogs.cpp:51]: Actually please revert to the original value when we use the same sentence in both places. Let's make less changes in the original code so it's easier to track.

**Comment by ihhub** [comment]
I'm okay with changing the color of titles in battle dialogs for now but let's don't change messages in this pull request.

File: src/fheroes2/battle/battle_dialogs.cpp [hunk 8]
```diff
@@ -389,8 +397,17 @@ void Battle::Arena::DialogBattleSummary( const Result & res ) const
...
-    TextBox box( msg, Font::BIG, 270 );
-    box.Blit( pos_rt.x + 25, pos_rt.y + 175 );
+    int32_t messageYOffset = 0;
+    if ( !title.empty() ) {
+        TextBox box( title, Font::BIG, bsTextWidth );
+        box.Blit( pos_rt.x + bsTextXOffset, pos_rt.y + bsTextYOffset );
+        messageYOffset = bsTextIndent;
+    }
+
+    if ( msg.length() > 0 ) {
```
Comment (ihhub) [inline src/fheroes2/battle/battle_dialogs.cpp:407]: Please use this `if ( !msg.empty() ) {`.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @shprotru , I left few comments. Could you please take a look?

File: src/fheroes2/battle/battle_dialogs.cpp (hunk 7 above)
Comment (shprotru) [reply src/fheroes2/battle/battle_dialogs.cpp]: done

File: src/fheroes2/battle/battle_dialogs.cpp (hunk 8 above)
Comment (shprotru) [reply src/fheroes2/battle/battle_dialogs.cpp]: done

**Comment by ihhub** [comment]
@shprotru thank a lot for fixing this issue!

**Pull request merged**